redaction/draft-strad-trans-redaction.md

```diff
@@ -69,26 +75,133 @@ certificates.
+  cases.
+
+* Therefore, we define a domain label redaction mechanism ({{redacting_labels}})
+  that covers all use cases, at the cost of considerably increased
+  implementation complexity.
+
+We anticipate that TLS clients may develop policies that impose additional
+compliancy requirements on the use of the {{name_constrained}} and
+{{redacting_labels}} mechanisms.
+
+To ensure effective redaction, CAs and domain owners should note the privacy
+considerations ({{privacy_considerations}}).
+
+TODO(eranm): Do we need to further expand (either here or in the following
+subsections) on when each of the mechanisms is/isn't suitable?
+
+## Using Wildcard Certificates    {#wildcard_certificates}
+
+A certificate containing a DNS-ID [RFC6125] of `*.example.com` could be used to
+secure the domain `topsecret.example.com`, without revealing the string
+`topsecret` publicly.
+
+Since TLS clients only match the wildcard character to the complete leftmost
+label of the DNS domain name (see Section 6.4.3 of [RFC6125]), a different
+mechanism is needed when any label other than the leftmost label in a DNS-ID is
+considered private (e.g., `top.secret.example.com`). Also, wildcard certificates
+are prohibited in some cases, such as Extended Validation Certificates
+[EV.Certificate.Guidelines].
+
+## Using a Name-Constrained Intermediate CA    {#name_constrained}
+
+An intermediate CA certificate or intermediate CA precertificate that contains
+the Name Constraints [RFC5280] extension MAY be logged in place of end-entity
+certificates issued by that intermediate CA, as long as all of the following
+conditions are met:
+
+* there MUST be a non-critical extension (OID 1.3.101.76, whose extnValue OCTET
+  STRING contains ASN.1 NULL data (0x05 0x00)). This extension is an explicit
+  indication that it is acceptable to not log certificates issued by this
+  intermediate CA.
+
+* there MUST be a Name Constraints extension, in which:
+
+  * permittedSubtrees MUST specify one or more dNSNames.
+
+  * excludedSubtrees MUST specify the entire IPv4 and IPv6 address ranges.
+
+Below is an example Name Constraints extension that meets these conditions:
+
+~~~~~~~~~~~
+    SEQUENCE {
+      OBJECT IDENTIFIER '2 5 29 30'
+      BOOLEAN TRUE
+      OCTET STRING, encapsulates {
+        SEQUENCE {
+          [0] {
+            SEQUENCE {
+              [2] 'example.com'
+              }
+            }
+          [1] {
+            SEQUENCE {
+              [7] 00 00 00 00 00 00 00 00
+              }
+            SEQUENCE {
+              [7]
+                00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+                00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+              }
+            }
+          }
+        }
+      }
+~~~~~~~~~~~
+
+### Presenting SCTs, Inclusion Proofs and STHs
+
+Each SCT (and optional corresponding inclusion proof and STH) presented by a TLS
+server MAY correspond to an intermediate CA certificate or intermediate CA
```

Comment:
Should we limit the number of name-constained CA certs in the chain to one?

That would effectively allow us to say if there's such a name-constrained CA cert (that meets the requirements in {{name_constrained}}), then it should be only one, and all SCTs observed in the connection (from other certs in the chain, TLS handshake or stapled OCSP response) refer to it.

That would extend the section in 6962-bis about SCT corresponding to the server certificate in a  deterministic  and less confusing way.